lib/portage/gpkg.py

```diff
@@ -1451,7 +1494,7 @@ def _load_manifest(self, manifest_string):
         manifest_filenames = []
 
         for manifest_record in manifest_string.splitlines():
-            if manifest_record == "":
+            if manifest_record == "" or manifest_record.startswith("[GNUPG:]"):
```

Comment:
Why would it contain `[GNUPG:]`?

Reply:
I don't know if this is a bug or expected behavior, but if GPG verification passes but the signature is untrusted or unknown, gpg will also send `[GNUPG:] XXX` to stdout instead of stderr.

Reply:
Weird. I've never seen it do that, and I see that you're using `--status-fd`, so it shouldn't happen really.

Reply:
Decrypt untrusted Manifest I got:
stderr:
```
 gpg: Signature made Thu Sep  1 10:06:15 2022 EDT
gpg:                using RSA key 8DEDA2CDED49C8809287B89D8812797DDF1DD192
gpg: Good signature from "Gentoo Portage Test Untrusted Key (Test Only, Do NOT Trust!!!) (Gentoo Test Key) <test@example.org>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 8DED A2CD ED49 C880 9287  B89D 8812 797D DF1D D192
```

stdout:
```
[GNUPG:] PLAINTEXT 74 0 
DATA gpkg-1 0 SHA512 cf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e BLAKE2B 786a02f742015903c6c6fd852552d272912f4740e15847618a86e217f71f5419d25e1031afee585313896444934eb04b903a685b1448b755d56f701afe9be2ce
DATA metadata.tar.zst 22 SHA512 489abee14a5c50af166e8f639fb9caf89cb871560c8d156bf782eede2bc8d2ab08515121015e67b061d7fb754892d6879cf71a622cc1c4e4eb3c4b0d97d841a6 BLAKE2B 7a3dfa9a6496afc90543e556d949bd77a7532716192b23711988104959e4d9f115b17f6ad8c8d5b817aac0e8d9b08b0024c62da3c08bd0b581aacacd40227686
DATA metadata.tar.zst.sig 833 SHA512 26d51d9fbca64e23394ac8940dbcb131581da63c2fe96d45d63df7fb81691cd9b2a592a28710fd14b6efcc450b93356e94b3384bb5527a3155cb562cb61ef2b6 BLAKE2B 2d2f3876abcb59b432367d4aa55012c7e028c8add40f9b3dffb8c2b87e11bc9da558e7e3b38da5eb68bc15841cc84d0ac9f76f72cfa1390546c83c751eca343e
DATA image.tar.zst 1049649 SHA512 97d6b3b4f38422c5ff048347e516199dab2238d50e2da83d31c46cd7df822870261e9d2c7f2485e0b9f88e3f4209c167670409f54adee20f081a7074f8ac105f BLAKE2B 6519051adbfe4d0077242a2313f5d70a975d20d95ceb306c2ebb496124471b5abe7a69f89f61411b878b9862d7a6fc6db15182e1db2f80575f322964671986ce
DATA image.tar.zst.sig 833 SHA512 543a9430b42a309f43767877f297ca74144442edaea4695e7d1c8f2822108a1c1665340c43a4b8b002b3498bf06dd8aa928c27c780fe3d73010d97eaeecfbf87 BLAKE2B 618474560052d74935e66c7db97efcb79aa9788c71e3aba9449299a81fd52ecbee8ab3863fb775a7e2c9464830a34c91c591367fa0be5b69c284b2f019601594
[GNUPG:] NEWSIG
[GNUPG:] KEY_CONSIDERED 8DEDA2CDED49C8809287B89D8812797DDF1DD192 0
[GNUPG:] SIG_ID uEoFdrEAEF2ZHJMgYTRU0qmBUKs 2022-09-01 1662041175
[GNUPG:] KEY_CONSIDERED 8DEDA2CDED49C8809287B89D8812797DDF1DD192 0
[GNUPG:] GOODSIG 8812797DDF1DD192 Gentoo Portage Test Untrusted Key (Test Only, Do NOT Trust!!!) (Gentoo Test Key) <test@example.org>
[GNUPG:] VALIDSIG 8DEDA2CDED49C8809287B89D8812797DDF1DD192 2022-09-01 1662041175 0 4 0 1 10 01 8DEDA2CDED49C8809287B89D8812797DDF1DD192
[GNUPG:] KEY_CONSIDERED 8DEDA2CDED49C8809287B89D8812797DDF1DD192 0
[GNUPG:] TRUST_UNDEFINED 0 pgp
```

This seems only happened when using cleartext signatures.

Reply:
Ok, I found what happened, `status-fd` wrong number in test case.  😂